src/tls/Extensions.fst

```diff
@@ -797,12 +832,12 @@ let rec extensionPayloadBytes = function
     | _                             -> vlbytes 2 (pskBytes psk))
   | E_early_data edi                -> vlbytes 2 (earlyDataIndicationBytes edi)
   | E_supported_versions vs         -> vlbytes 2 (protocol_versions_bytes vs)
-  | E_cookie c                      -> (lemma_repr_bytes_values (length c); vlbytes 2 (vlbytes 2 c))
+  | E_cookie c                      -> (lemma_repr_bytes_values (length c); vlbytes 2 c)
```

Comment:
The formatting of `E_cookie c` depends on whether `c` is a raw cookie or is already length-encoded. If it's a raw cookie, we need to first prepend its length (2 bytes), and then prepend 2 more bytes to the result for the length of the extension data.
I believe the previous encoding was correct as `c` comes from `Ticket.create_cookie`, which is an encrypted blob. When a client receives this blob, it will interpret the first two bytes as the length and either fail, or echo back some garbage.

```
struct {
          opaque cookie<1..2^16-1>;
      } Cookie;

struct {
       ExtensionType extension_type;
       opaque extension_data<0..2^16-1>;
   } Extension;
````